users/src/main/java/io/crate/operation/user/UsersMetaData.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.operation.user;
+
+import org.elasticsearch.cluster.AbstractDiffable;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.*;
+
+public class UsersMetaData extends AbstractDiffable<MetaData.Custom> implements MetaData.Custom {
+
+    public static String TYPE = "users";
+
+    static final UsersMetaData PROTO = new UsersMetaData();
+
+    private final List<String> users;
+
+    private UsersMetaData() {
+        this.users = new ArrayList<>();
+    }
+
+    private UsersMetaData(List<String> users) {
+        this.users = users;
+    }
+
+    public static UsersMetaData newInstance(UsersMetaData instance) {
+        return new UsersMetaData(new ArrayList<>(instance.users));
+    }
+
+    static UsersMetaData of(String... user) {
```

Comment:
Can we remove this factory method and only have a constructor that takes the expected data structures without providing a convenience function that involves unnecessary transformations?